packages/ember-metal/lib/weak_map.js

```diff
@@ -115,7 +115,7 @@ WeakMap.prototype.has = function(obj) {
  */
 WeakMap.prototype.delete = function(obj) {
   if (this.has(obj)) {
-    delete metaFor(obj).writableWeak()[this._id];
```

Comment:
`this.has` checks with `peekMeta` already so should be peekMeta here too
https://github.com/bekzod/ember.js/blob/14243438756b908b4557f9f3e296d3af89bcda81/packages/ember-metal/lib/weak_map.js#L100